runtime/jvmti/jvmtiClass.c

```diff
@@ -1172,104 +1172,104 @@ redefineClassesCommon(jvmtiEnv* env,
 		/* Recreate the RAM classes for all classes */
 
 		rc = recreateRAMClasses(currentThread, classPairs, methodPairs, extensionsUsed, !extensionsEnabled);
-		if (rc == JVMTI_ERROR_NONE) {
+		if (rc != JVMTI_ERROR_NONE) {
+			goto failedWithVMAccess;
```

Comment:
The hash table is not freed along this path. It looks like it could just be moved to after the label.